clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/HelixScatterGatherRoutingStrategy.java

```diff
@@ -36,7 +40,7 @@ public List<String> getReplicas(long requestId, List<String> replicas, int requi
         if (selectedReplicas.size() == requiredReplicaCount) {
           return selectedReplicas;
         }
-        if (helixGroupInfo.get(replica) == groupId) {
+        if (helixGroupInfo.get(replica) == groupId && !instanceHealthMonitor.isInstanceBlocked(replica)) {
```

Comment:
No, that'd defeat some purpose of Helix assisted routing. If we do want to add some notion of least loaded we also need to make some changes since HAR is mostly going to be used for batch gets and currently the instance monitor only tracks request not key count.